### PR TITLE
Time: Change old `u32` timestamps to 64-bit

### PR DESCRIPTION
**Pull request by SmallJoker**
Main problem reported by VanessaE, where there are timeout issues in the connection.

### CODE REVIEW

File: src/clientiface.cpp [hunk 1]
```diff
@@ -590,9 +590,9 @@ void RemoteClient::notifyEvent(ClientStateEvent event)
 	}
 }
 
-u32 RemoteClient::uptime() const
+u64 RemoteClient::uptime() const
 {
-	return porting::getTime(PRECISION_SECONDS) - m_connection_time;
+	return porting::getTimeMs() - m_connection_time;
```
Comment (nerzhul) [inline src/clientiface.cpp]: precision_seconds not Ms

**Comment by Fixer-007** [comment]
Should fix this bug https://github.com/minetest/minetest/issues/5799

File: src/clientiface.cpp (hunk 1 above)
Comment (nerzhul) [inline src/clientiface.cpp]: -> getTimeS

**Comment by nerzhul** [comment]
I found the issue, big WTF, there is two constructor, one with std::string, the other with const char* and doing a std::string operation on the const char *. I just don't understand why it works on Linux & Windows, it's crazy. Fixing it

**Comment by nerzhul** [comment]
pffff... MacOSX refused to build another time... if someone has an idea on how to fix permitting to merge... @Zeno- @sfan5 @SmallJoker

**Comment by nerzhul** [comment]
It's now fixed. Ready for merge

**Comment by SmallJoker** [comment]
MSVC also compiled it just fine.

**Pull request merged**